### PR TITLE
Added Negative Axes support along with TCs.

### PR DESCRIPTION
**Pull request by amitsrivastava78**
Added the support for negative axes along with TCs

### CODE REVIEW

File: tensorflow/lite/kernels/reverse_test.cc [hunk 1]
```diff
@@ -77,6 +77,48 @@ TEST(ReverseOpTest, FloatMultiDimensions) {
                         17, 18, 15, 16, 13, 14, 23, 24, 21, 22, 19, 20}));
 }
 
+TEST(ReverseOpTest, FloatMultiAxes) {
+  ReverseOpModel<float> model({TensorType_FLOAT32, {4, 3, 2}},
+                              {TensorType_INT32, {1}});
+  model.PopulateTensor<float>(model.input(),
+                              {1,  2,  3,  4,  5,  6,  7,  8,  9,  10, 11, 12,
+                               13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24});
+  model.PopulateTensor<int32_t>(model.axis(), {2});
```
Comment (renjie-liu) [inline tensorflow/lite/kernels/reverse_test.cc]: why call this multiaxes? it seems it's still one axis?

Comment (amitsrivastava78) [reply tensorflow/lite/kernels/reverse_test.cc]: @renjie-liu , sorry for the oversight, i have updated the name of the TC, kindly check and approve.

**Comment by amitsrivastava78** [comment]
@renjie-liu , sorry for the oversight, i have updated the name of the TC, kindly check and approve.

File: tensorflow/lite/kernels/reverse_test.cc [hunk 2]
```diff
@@ -77,6 +77,48 @@ TEST(ReverseOpTest, FloatMultiDimensions) {
                         17, 18, 15, 16, 13, 14, 23, 24, 21, 22, 19, 20}));
 }
 
+TEST(ReverseOpTest, FloatAxes) {
```
Comment (renjie-liu) [inline tensorflow/lite/kernels/reverse_test.cc]: nit: axis

**Review by renjie-liu — CHANGES_REQUESTED** [review]

File: tensorflow/lite/kernels/reverse_test.cc (hunk 2 above)
Comment (amitsrivastava78) [reply tensorflow/lite/kernels/reverse_test.cc]: @renjie-liu , thanks for the comments, i have updated as suggested by you.Kindly check and approve.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@renjie-liu , thanks for the comments, i have updated as suggested by you.Kindly check and approve.

Regards
Amit

**Review by renjie-liu — DISMISSED** [review]

**Comment by amitsrivastava78** [comment]
@rthadur , can you pls help to get this merged, as this is approved and all checks have passed  and i am afraid that if it stays longer again some merge conflicts might come.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@rthadur , this PR is approved, for long, can you please help me to get this merged.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@renjie-liu , thanks for approving the PR, this PR is not yet merge, can you please help me to get this merge.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@rthadur , can you please help to get this merged.

Regards
Amit

**Comment by rthadur** [comment]
working on it, thank you for your patience.

File: tensorflow/lite/kernels/reverse.cc [hunk 3]
```diff
@@ -69,7 +69,11 @@ TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
   const TfLiteTensor* axis_tensor = GetInput(context, node, kAxisTensor);
   int axis = GetTensorData<int32_t>(axis_tensor)[0];
 
-  TF_LITE_ENSURE(context, axis >= 0 && axis < NumDimensions(input));
+  if (axis < 0) {
+    axis += input->dims->size;
+  }
+  TF_LITE_ENSURE_MSG(context, axis >= 0 && axis < NumDimensions(input),
```
Comment (jianlijianli) [inline tensorflow/lite/kernels/reverse.cc]: Nit: it seems the axis >= 0 check is no longer needed with the current change.

**Review by jianlijianli — DISMISSED** [review]

**Comment by amitsrivastava78** [comment]
@jianlijianli , thanks for approving the PR

@rthadur , can you please help to merge this PR.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@rthadur , can you please help to merge this PR.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@rthadur , can you please help to meger this PR.


Regards
Amit

File: tensorflow/lite/kernels/reverse_test.cc [hunk 4]
```diff
@@ -77,6 +77,48 @@ TEST(ReverseOpTest, FloatMultiDimensions) {
...
+}
+
+TEST(ReverseOpTest, FloatInvalidAxes) {
+  ReverseOpModel<float> model({TensorType_FLOAT32, {4, 3, 2}},
+                              {TensorType_INT32, {1}});
+  model.PopulateTensor<float>(model.input(),
+                              {1,  2,  3,  4,  5,  6,  7,  8,  9,  10, 11, 12,
+                               13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24});
+  model.PopulateTensor<int32_t>(model.axis(), {-4});
+  EXPECT_DEATH(model.Invoke(), "Invalid Axes passed");
```
Comment (rthadur) [inline tensorflow/lite/kernels/reverse_test.cc]: tensorflow/lite/kernels/reverse_test.cc:119: Failure
Death test: model.Invoke()
    Result: failed to die.
 Error msg:
[  DEATH   ] ERROR: tensorflow/lite/kernels/reverse.cc Invalid Axes passed
[  DEATH   ] ERROR: Node number 0 (REVERSE_V2) failed to invoke.
[  DEATH   ] 
[  DEATH   ] 
[  FAILED  ] ReverseOpTest.FloatInvalidAxes (19 ms)

**Review by rthadur — CHANGES_REQUESTED** [review]
there are internal tests failing , please check

File: tensorflow/lite/kernels/reverse_test.cc (hunk 4 above)
Comment (amitsrivastava78) [reply tensorflow/lite/kernels/reverse_test.cc]: @rthadur the PR was quite old and the test framework changes lead to this failure, i have modified the file and now everything seems passing.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@renjie-liu , @jianlijianli @rthadur , i have updated the code and now everything seems to be passing, kindly check and approve, hope to see this PR merge soon.

Regards
Amit

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by gbaned** [comment]
@amitsrivastava78 Can you please resolve conflicts? Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by gbaned** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!

**Pull request closed without merging**